x-pack/plugins/cloud_security_posture/README.md

```diff
@@ -4,6 +4,84 @@ Cloud Posture automates the identification and remediation of risks across cloud
 
 ---
 
+## Table of contents
+
+- [Development](#development)
+  - [Install pre-commit hooks](#install-pre-commit-hooks)
+  - [Local checks before creating a PR](#local-checks-before-creating-a-pr)
+  - [Running unit tests](#running-unit-tests)
+  - [Running integration tests](#running-integration-tests)
+
+---
+
 ## Development
 
-See the [kibana contributing guide](https://github.com/elastic/kibana/blob/main/CONTRIBUTING.md) for instructions setting up your development environment.
+See the [kibana contributing guide](https://github.com/elastic/kibana/blob/main/CONTRIBUTING.md) for instructions
+setting up your development environment.
+
+### Install pre-commit hooks
+
+We
+use [pre-commit](https://docs.elastic.dev/kibana-dev-docs/getting-started/setup-dev-env#install-pre-commit-hook-optional)
+to run linters and tests before each commit. To install the pre-commit hooks, run the following command from the root of
+the repository:
+
+```bash
+node scripts/register_git_hook
+```
+
+### Local checks before creating a PR
```

Comment:
pre-commits hooks and local checks before creating PR are kinda the same?

i'd include just our own specific checks. the hooks are optional and not required for development.

Reply:
personally i find them more annoying than useful tbh. i do many commits, some just to mark where i am, and don't care about checks at that moment, and with pre-hooks you need to pass `--no-verify` which isn't nice. 

i'm also not aware of others in our team using it. in anyway, all i meant is that the emphasis we put in our readme (first thing to do - install prehooks) isn't actually required (or commonly used)

Reply:
Good point.

I changed the order of titles + I added it as `optional` step